File: Unit 1 Fundamentals of programming/Fundamentals Topic 1 Programming Basics/Topic 1 VB Programs/W1 Q2 Car Mileage Per Gallon.vb
Module Module1

    Sub main()
        'W1 Q2 Car Mileage Per Gallon

        'declare variable and assign values returned from input boxes

        Dim lastCarMileage As Integer = InputBox("Enter last car milage")
        Dim currentCarmileage As Integer = InputBox("Enter current car milage")
        Dim totalMiles As Integer = currentCarmileage - lastCarMileage
        Dim litresToFillTank As Single = InputBox("Enter litres to fill tank")

        'Declare variables and assign values returned from calculations
        Dim gallonsToFillTank As Single = litresToFillTank / 4.546
        Dim mileage As Single = totalMiles / gallonsToFillTank

        'Concatenate string with variable value using &
        MsgBox("Mileage: " & mileage)

    End Sub

End Module

File: A-Level/Unit 1 Fundamentals of programming/Fundamentals Topic 1 Programming Basics/Topic 1 Python Programs/W1 Q2 Car mileage per gallon.py
#AQA AS W1 Q2 Car mileage per gallon

litresPerGallon = 4.546
previousCarMileage = float(input("Enter previous mileometer reading: "))
currentCarMileage = float(input("Enter current mileometer reading: "))
litresToFillTank = float(input("Enter litres needed to fill tank: "))
totalMiles = currentCarMileage - previousCarMileage
gallonsToFillTank = litresToFillTank / litresPerGallon
mileage = totalMiles /gallonsToFillTank
mileage = int(mileage)
print("Miles per gallon :", mileage)
